### PR TITLE
Upmerge 14.12.2020

### PR DESCRIPTION
**Pull request by de-nordic**
[nrf mergeup] Merge upstream zephyr up to commit 856fd652da

Regular upmerge. Synchronizing sdk-zephyr
with zephyrproject-rtos/zephyr/master.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

**Comment by edmont** [comment]
@de-nordic could we get OpenThread upmerge in as well? https://github.com/zephyrproject-rtos/zephyr/pull/30745

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Squashing reverts like this breaks west ncs-loot, which breaks the rebase procedure we use for releases.

Please follow the upmerge procedure as documented by using individual revert commits.

**Comment by de-nordic** [comment]
Please cease with reviews as I am going to follow @edmont request and move to upstream/master that will include zephyrproject-rtos/zephyr#30745. Thanks!

**Review by mbolivar-nordic — APPROVED** [review]
Structure looks fine; `west ncs-loot -z 856fd652da zephyr` says:

```
NCS commit (manifest-rev): 4ebcc59f83f36f642eaf78c95c38285c9ede366c, upstream commit: 856fd652dab926874131c4fb8ecd985ee1e297be
OOT patches: 22 total
- 39f7253a986e3b8cb9ef0c8935a2b32c59d8b5f4 [nrf noup] bluetooth: controller: default to SPLIT architecture
- 19c66e05d970234e84f8d8571e0e33a8a5ba7fe4 [nrf noup] net: lib: mqtt: Provide option to enable TLS ession caching
- 45958693f28fb7cdf0b5414e10b9ce7e2a81ce59 [nrf noup] dts: add compatibles for CC310 NS access
- 9c42b9b13d974b7a56642004544b2fc588099693 [nrf noup] dts: prefer CC310 for zephyr,entropy
- 324b6fbb139a69a366ae0f884bda589c9e8e4ffa [nrf noup] codeowner: Add entry for .github and Jenkinsfile
- 23df4b66d44ecc20974598c41bb095995e12def0 [nrf noup] ci: Add Jenkins-based CI for zephyr
- 1a64079c8fe3c0b42f5888abe0d774ec2cb74fba [nrf noup] ci: move jenkinsfile to test-ci-nrfconnect-cfg
- 373acf80e0a3a743e9bfc9ab5d0e8a4e6a56c6b1 [nrf noup] net: l2: Kconfig: Add option for Zigbee L2 layer
- d875660a86418497a90e12185914eae902105e4c [nrf toup] drivers: clock_control: nrf: Change CLOCK_CONTROL_NRF_FORCE_A
- 7c4e6bb7958a62e4666fde529c6f7e77d6a44d04 [nrf toup] drivers: clock_control: nrf: Add assert on unexpected event
- a32dc3c16743280c5b4042a5b3b41034631b78e4 [nrf toup] drivers: clock_control: nrf: Add CLOCK_CONTROL_NRF_FORCE_ALT
- 0f3124de75a0159c5011ef520cc3fe7d47ed37f6 [nrf toup] drivers: Adding cryptocell as chosen entropy for nRF5340
- 06f3f56d5dc48e26d8b5937ac9167fd3f536f569 [nrf temphack] soc: use PM when configuring MPU regions
- 08f6b201d623425db6cc35193643a6d814919f8e [nrf noup] include: net: add NCS extensions
- 14b877dc780c06838059e1bea1cdda6fc5d6d7f3 [nrf noup] tree-wide: support NCS Partition Manager (PM) definitions
- 63d7b52f82229c5d7c487fa1e04fb7542286cc8a [nrf noup] linker: use partition manager to set RAM
- 6261e168e31c3a780db07d1fbe6fa080ce7dfa8f [nrf noup] doc: Share Kconfig documentation between repos
- 800654d7de7af8c15ca7d057547bba01c4cd4b12 [nrf noup] net: lib: mqtt: Transition to zsock_setsockopt
- 69a43b4442834c886b46f215fa1b7d3a5c822b69 [nrf fromlist] Bluetooth: Mesh: Fix friend node establishment and message
- 6f6634087389f45c7d757b1ca3df76306a334c0b [nrf temphack] runners: nrfjprog: hackaround NCSDK-7327
- 562c2c431fcaafff65dea035c44e62a9777eb265 [nrf noup] drivers: ieee802154: Use nRF Secure Services to read EUI64
- 4ebcc59f83f36f642eaf78c95c38285c9ede366c [nrf noup] drivers: flash: No spm_request_read when no secure services
```

Cherry-picking the OOT patches onto 856fd652dab926874131c4fb8ecd985ee1e297be yields this clean rebase, for reference:

https://github.com/mbolivar-nordic/fw-nrfconnect-zephyr/tree/upmerge-413-rebase

Nice work.

**Review by rlubos — APPROVED** [review]
LGTM

**Comment by de-nordic** [comment]
> 
> Cherry-picking the OOT patches onto [856fd65](https://github.com/nrfconnect/sdk-zephyr/commit/856fd652dab926874131c4fb8ecd985ee1e297be) yields this clean rebase, for reference:
> 
> https://github.com/mbolivar-nordic/fw-nrfconnect-zephyr/tree/upmerge-413-rebase
> 
> Nice work.

Thanks for verifying!

**Comment by mbolivar-nordic** [comment]
@de-nordic can you please cherry-pick https://github.com/zephyrproject-rtos/zephyr/pull/31092 as a fromlsit?

**Review by trond-snekvik — APPROVED** [review]

File: boards/arm/nrf5340dk_nrf5340/Kconfig.defconfig [hunk 1]
```diff
@@ -20,7 +20,8 @@ config BOARD
 # force building with TF-M as the Secure Execution Environment.
 
 config BUILD_WITH_TFM
-	default y if BOARD_NRF5340DK_NRF5340_CPUAPPNS || BOARD_NRF5340PDK_NRF5340_CPUAPPNS
+        default n
```
Comment (ioannisg) [inline boards/arm/nrf5340dk_nrf5340/Kconfig.defconfig:23]: Could you add a comment for this temp-hack?
Something around this lines 
`Temporarily disable building Non-Secure images with TF-M support by default.`
also, might be an indentation issue here.

Comment (ioannisg) [reply boards/arm/nrf5340dk_nrf5340/Kconfig.defconfig:23]: @oyvindronningstad @anangl , pls, check this out

**Review by tejlmand — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

File: boards/arm/nrf5340dk_nrf5340/Kconfig.defconfig (hunk 1 above)
Comment (anangl) [reply boards/arm/nrf5340dk_nrf5340/Kconfig.defconfig]: > One alternative option is to remove the whole Kconfig definition of BUILD_WITH_TFM from this file.

I prefer the current form, as it makes the presence of this temp-hack more apparent.

**Review by anangl — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Pull request merged**